src/core/internal/modes/dialog.ts

```diff
@@ -735,16 +739,18 @@ export function dialog(parameters: dialog.Parameters = {}) {
         const feeToken = await resolveFeeToken(internal, parameters)
 
         // Try and extract an authorized key to sign the calls with.
-        const key = await Mode.getAuthorizedExecuteKey({
-          account,
-          calls,
-          permissionsId: parameters.permissionsId,
-        })
+        const key =
+          account &&
+          (await Mode.getAuthorizedExecuteKey({
+            account,
+            calls,
+            permissionsId: parameters.permissionsId,
+          }))
```

Comment:
Prefer narrowing the type of `key` to the semantic type instead of potentially `false`.

```suggestion
          account ?
          (await Mode.getAuthorizedExecuteKey({
            account,
            calls,
            permissionsId: parameters.permissionsId,
          })) : undefined
```